service/start.go

```diff
@@ -119,6 +115,127 @@ func (svc *service) startNostr(ctx context.Context, encryptionKey string) error
 	return nil
 }
 
+type createAppSubscriber struct {
+	events.EventSubscriber
+	svc   *service
+	relay *nostr.Relay
+}
+
+func (s *createAppSubscriber) ConsumeEvent(ctx context.Context, event *events.Event, globalProperties map[string]interface{}) {
+	if event.Event != "app_created" {
+		return
+	}
+	properties, ok := event.Properties.(map[string]interface{})
+	if !ok {
+		logger.Logger.WithField("event", event).Error("Failed to cast event.Properties to map")
+		return
+	}
+	id, ok := properties["id"].(uint)
+	if !ok {
+		logger.Logger.WithField("event", event).Error("Failed to get app id")
+		return
+	}
+	walletPrivKey, err := s.svc.keys.GetAppWalletKey(uint32(id))
+	if err != nil {
+		logger.Logger.WithError(err).Error("Failed to calculate app wallet priv key")
+	}
+	walletPubKey, _ := nostr.GetPublicKey(walletPrivKey)
```

Comment:
can the error be caught here?  rather than checking the walletPubKey is not "" a few lines below